packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm

```diff
@@ -91,6 +91,15 @@ - (void)setBackgroundColor:(UIColor *)backgroundColor
   _backgroundColor = backgroundColor;
 }
 
+- (void)traitCollectionDidChange:(UITraitCollection *)previousTraitCollection
+{
+  [super traitCollectionDidChange:previousTraitCollection];
+
+  if ([self.traitCollection hasDifferentColorAppearanceComparedToTraitCollection:previousTraitCollection]) {
+    [self invalidateLayer];
+  }
+}
```

Comment:
I'm not sure you have to implement this, if you create your colors with the `colorWithDynamicProvider`. This is a closure that should update the color when the traitCollection changes automatically. 🤔 
What happens if we remove this? Have you tried?

Reply:
We use the layer to render background color, and it is CGColor, so we need to call invalidate programmally.